File: apps/cove-gateway/src/composition/bootstrap.ts
import {
  listenPort,
  loadRuntimeFromEnv,
} from "../features/runtime/infrastructure/env-runtime";
import { startOtel } from "../features/telemetry/infrastructure/start-otel";
import { createJsonIdentityStore } from "../features/identity/infrastructure/json-identity-store";
import type { IdentityStore } from "../features/identity/application/ports/identity-store";

export function bootstrap(defaultPort: number) {
  const ctx = loadRuntimeFromEnv("cove-gateway");
  const telemetry = startOtel(ctx);
  const port = listenPort(defaultPort);
  const identityEnabled = Bun.env.IDENTITY_ENABLED === "true";
  const identity: IdentityStore = createJsonIdentityStore({
    enabled: identityEnabled,
    seedPath: Bun.env.IDENTITY_SEED_PATH ?? "/data/identity/seed.json",
  });
  return { ctx, telemetry, port, identity };
}

File: apps/cove-workers/Sources/cove_workers/Composition/Bootstrap.swift
import Foundation

#if canImport(Glibc)
import Glibc
#elseif canImport(Darwin)
import Darwin
#endif

enum Bootstrap {
  static func run(service: String, defaultPort: UInt16) -> (
    RuntimeContext, Telemetry, UInt16, IdentityStore
  ) {
    let ctx = EnvRuntime.load(service: service)
    let tel = Telemetry(ctx: ctx)
    let port = EnvRuntime.port(fallback: defaultPort)
    let enabled = ProcessInfo.processInfo.environment["IDENTITY_ENABLED"] == "true"
    let seed =
      ProcessInfo.processInfo.environment["IDENTITY_SEED_PATH"]
      ?? "/data/identity/seed.json"
    let identity = IdentityStore(enabled: enabled, seedPath: seed)
    return (ctx, tel, port, identity)
  }
}
